### PR TITLE
feat: Add a condition to evaluate valid scenarios when using whitespace as delimiter

### PR DESCRIPTION
**Pull request by favipcj**
* add a condition to evaluate valid scenarios when using whitespace as delimiter like:  `'"agroup" "delimited by" "spaces"'` 
* add test cases for white space delimiter

### CODE REVIEW

**Comment by dfresh613** [comment]
This should cover the last of the corner cases we have seen with certain oidc clients passing headers from openldap. 

As evidenced by additional unit test scenarios, this helpful change allows support for a comma delimited set of groups within a single header string.

Once this is approved and merged, can we please make a new release of flask_authz (probably 2.3.1)?

**Comment by jessecooper** [comment]
lgtm;

**Comment by favipcj** [comment]
> lgtm, but looks like the build has some test failing. Will need to look closer at the issues in those test runs.

TI seems to be because of the last version of  `casbin_sqlalchemy_adapter==0.4.0`  https://github.com/pycasbin/flask-authz/pull/36#issuecomment-884186374

**Comment by jessecooper** [comment]
Please do open that additional PR to fix the issue with `casbin_sqlalchemy_adapter` with that said I dont see any reason to block this PR on that. lgtm;

**Comment by favipcj** [comment]
FYI. I'm just pinning casbin_sqlalchemy_adapter==0.3.2 for now, as there are some changes on the last version of `pycasbin` https://github.com/casbin/pycasbin/commit/d69a2dfa7711aebcf7cec880fa724f5d3b3d3944 that are allowing to import `UpdateAdapter` directly from `persist.adapters`, but those changes were not taken to run the tests successfully on `flask-authz` as this package is still using `casbin==1.0.4.`

**Comment by ffyuanda** [comment]
lgtm

File: tests/test_casbin_enforcer.py
Comment (ffyuanda) [inline tests/test_casbin_enforcer.py:166]: Why is this part here? I believe we already have `test_sanitize_group_headers` at line 239

File: tests/test_casbin_enforcer.py
Comment (ffyuanda) [inline tests/test_casbin_enforcer.py:156]: This line should not be OK coz we use ',' as delimiter here

**Comment by hsluoyz** [comment]
@techoner plz review

File: tests/test_casbin_enforcer.py
Comment (favipcj) [reply tests/test_casbin_enforcer.py]: I missed this.. thanks for pointing it out, it's deleted now and as this was a duplicated test, it was not running at all, so that is why the test you pointed out in your other comment was not failing.

**Review by ffyuanda — APPROVED** [review]
lgtm;

**Review by jessecooper — APPROVED** [review]
lgtm;

**Comment by jessecooper** [comment]
Going to merge, will bump the version to 2.3.1 as soon as I can unless someone gets to it before I do.

**Pull request merged**
